src/server/resc_attr.c

```diff
@@ -994,35 +1031,50 @@ action_min_walltime(resource *presc, attribute *pattr, void *pobject, int type,
 			return PBSE_NOSTF_JOBARRAY;
 #endif
 		/* STF jobs can't request soft_walltime */
-		def = find_resc_def(svr_resc_def, SOFT_WALLTIME, svr_resc_size);
-		if (!def)
+		if (soft_walltime_def == NULL)
+			soft_walltime_def = find_resc_def(svr_resc_def, SOFT_WALLTIME, svr_resc_size);
+		if (soft_walltime_def == NULL)
```

Comment:
I'm not sure if you want to return from here... Although it is almost impossible to have a NULL return for these in-built resources but the action function is for min_walltime. I think we should continue forward to compare it with max_walltime